### DISCUSSION TITLE
Does GPTCache support storing retrived documents as well as LLM answers?

### DISCUSSION THREAD

**Discussion by renuka010** [zilliztech/GPTCache#629] [Q&A]
I understood how we store query and LLM responses in GPTCache. I want to cache retrieved documents along this. I saw similar example here https://gptcache.readthedocs.io/en/latest/bootcamp/langchain/question_answering.html

llm = LangChainLLMs(llm=OpenAI(temperature=0))
chain = load_qa_chain(llm, chain_type="stuff")
query = "What did the president say about Justice Breyer"
chain.run(input_documents=docs, question=query)

 But I had to perform similarity search only for query and able to get answer along with documents in response. Is that possible with GPTCache?

**Comment by SimFG** [comment]
It is recommended to set cache through langchain, because LangChainLLMs cannot keep up with the update of langchain.
refer to: https://github.com/zilliztech/GPTCache/issues/585#issuecomment-1972720103

**Reply by renuka010** [reply]
Got it. Thanks.
